File: functions/libsodium/variants/go/4_fBlaMka_9.go
package main

func fBlaMka(x, y uint64) uint64 {
	const mask uint64 = 0xFFFFFFFF
	prod := (x & mask) * (y & mask)
	return (2 * prod) + x + y
}


func main(){
    fBlaMka(0,0)
}

File: functions/libsodium/variants/c/4_fBlaMka_9.c
#include <stdint.h>

static inline uint64_t fBlaMka(uint64_t x, uint64_t y)
{
    const uint64_t mask = UINT64_C(0xFFFFFFFF);
    const uint64_t product = (x & mask) * (y & mask);
    return x + y + 2 * product;
}


int main(){
    fBlaMka(0,0);
}